subsys/bluetooth/controller/ll_sw/ull_filter.c

```diff
@@ -73,15 +74,45 @@ static struct rl_dev {
 	bt_addr_t curr_rpa;
 	bt_addr_t peer_rpa;
 	bt_addr_t *local_rpa;
-
+#if defined(CONFIG_BT_CTLR_SW_DEFERRED_PRIVACY)
+	bt_addr_t target_rpa;
+#endif
 } rl[CONFIG_BT_CTLR_RL_SIZE];
 
+#if defined(CONFIG_BT_CTLR_SW_DEFERRED_PRIVACY)
+/* Cache of known unknown peer RPAs */
+static u8_t newest_prpa;
+static struct prpa_cache_dev {
+	u8_t      taken:1;
+	bt_addr_t rpa;
+} prpa_cache[CONFIG_BT_CTLR_RPA_CACHE_SIZE];
+
+struct prpa_resolve_work {
+	struct k_work prpa_work;
+	bt_addr_t     rpa;
+	resolve_callback_t cb;
+};
+
+struct target_resolve_work {
+	struct k_work target_work;
+	bt_addr_t rpa;
+	u8_t      idx;
+	resolve_callback_t cb;
+};
+#endif /* CONFIG_BT_CTLR_SW_DEFERRED_PRIVACY */
+
 static u8_t peer_irks[CONFIG_BT_CTLR_RL_SIZE][16];
 static u8_t peer_irk_rl_ids[CONFIG_BT_CTLR_RL_SIZE];
 static u8_t peer_irk_count;
 
 static bt_addr_t local_rpas[CONFIG_BT_CTLR_RL_SIZE];
 
+#if defined(CONFIG_BT_CTLR_SW_DEFERRED_PRIVACY)
+static struct prpa_resolve_work resolve_work;
+static struct target_resolve_work t_work;
+
+BUILD_ASSERT(ARRAY_SIZE(prpa_cache) < FILTER_IDX_NONE);
+#endif /* CONFIG_BT_CTLR_SW_DEFERRED_PRIVACY */
 BUILD_ASSERT(ARRAY_SIZE(wl) < FILTER_IDX_NONE);
 BUILD_ASSERT(ARRAY_SIZE(rl) < FILTER_IDX_NONE);
```

Comment:
Maybe one assert is enough?
CONFIG_BT_CTLR_RL_SIZE < FILTER_IDX_NONE?

Reply:
All 3 different arrays,one could argue if these asserts serve a valuable purpose. In any case I have simply followed the convention that was pre-existing in the file.